Sources/NIOCore/NIOScheduledCallback.swift

```diff
@@ -170,3 +172,23 @@ extension EventLoop {
         }
     }
 }
+
+@usableFromInline
+struct LoopBoundScheduledCallbackHandlerWrapper<Handler: NIOScheduledCallbackHandler>: NIOScheduledCallbackHandler, Sendable {
+    private let box: NIOLoopBound<Handler>
+
+    @usableFromInline
+    init(wrapping handler: Handler, eventLoop: some EventLoop) {
+        self.box = .init(handler, eventLoop: eventLoop)
+    }
+
+    @usableFromInline
+    func handleScheduledCallback(eventLoop: some EventLoop) {
+        self.box.value.handleScheduledCallback(eventLoop: eventLoop)
+    }
+
+    @usableFromInline
+    func didCancelScheduledCallback(eventLoop: some EventLoop) {
+        self.box.value.didCancelScheduledCallback(eventLoop: eventLoop)
+    }
```

Comment:
This type is generic over a user provided `Handler`, to ensure perf, this should not just be `@useableFromInline` it should be `@inlinable` instead.
```suggestion
    @inlinable
    init(wrapping handler: Handler, eventLoop: some EventLoop) {
        self.box = .init(handler, eventLoop: eventLoop)
    }

    @inlinable
    func handleScheduledCallback(eventLoop: some EventLoop) {
        self.box.value.handleScheduledCallback(eventLoop: eventLoop)
    }

    @inlinable
    func didCancelScheduledCallback(eventLoop: some EventLoop) {
        self.box.value.didCancelScheduledCallback(eventLoop: eventLoop)
    }
```